### DISCUSSION TITLE
Why such old compilers??

### DISCUSSION THREAD

**Discussion by lightyear15** [roc-streaming/roc-toolkit#261] [Q&A] [answered]
I would like start contributing to Roc, it looks like a human-sized codebase, where one can practice some C++. However compilers supported are so old that I would basically have to code in C++03.
Is there any reason for supporting such old compilers?
Even avr-gcc , mips-gcc and msp430-gcc are newer than the targeted gcc version

**Comment by gavv** [comment, marked as answer]
There are two reasons:

* The most of the code base is intentionally limited to a *subset* of C++98. You can think about it as a kind of "C with classes". You can read the details in [our documentation](https://roc-project.github.io/roc/docs/development/coding_guidelines.html). This approach is quite different from modern C++. We're trying to keep the code simple and straightforward.

  (Note that I'm not saying modern C++ is bad. I'm just saying that in this project, a different approach was chosen.)

* I used to deal with (very) outdated compilers at work, which is usual for various Linux- and non-Linux embedded systems. Often compilers come from the hardware vendor and old boards never get updates. And since, from one hand, we want to be portable, and on the other hand, we don't need C++11/14/17 support anyway, we can afford supporting the old compilers too, unless it will burden us too much.

**Comment by lightyear15** [comment]
Alright,
thanks for the answer.
